diff --git a/qa/common.js b/qa/common.js
index fffa522..f6a592c 100644
--- a/qa/common.js
+++ b/qa/common.js
@@ -28,13 +28,13 @@ function setupMediaRecorder(stream, numberOfRecorders, mimeType) {
 
   for(var i = 0; i < numberOfRecorders; i++){
     var mediaRecorder = new MediaRecorder(stream);
-    var blobDataAvailable = [];
+    mediaRecorder.blobdata = [];
 
     mediaRecorder.ondataavailable = function(evt) {
       console.log('ondataavailable fired');
       console.log(evt);
       console.log(mediaRecorderAttributeDump(evt.target));
-      blobDataAvailable.push(evt.data);
+      evt.target.blobdata.push(evt.data);
     };
 
     mediaRecorder.onerror = function(evt) {
@@ -47,8 +47,8 @@ function setupMediaRecorder(stream, numberOfRecorders, mimeType) {
       console.log('onstop fired');
       console.log(evt);
       console.log(mediaRecorderAttributeDump(evt.target));
-      updateBlobURLUI(new Blob(blobDataAvailable, { 'type' : mimeType }));
-      blobDataAvailable = [];
+      updateBlobURLUI(new Blob(evt.target.blobdata, { 'type' : mimeType }));
+      evt.target.blobdata = [];
     };
 
     mediaRecorder.onwarning = function(evt) {

